CMakeLists.txt

```
@@ -0,0 +1,36 @@
+################################################################################
+#    Copyright (C) 2021 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    #
+#                                                                              #
+#              This software is distributed under the terms of the             #
+#              GNU Lesser General Public Licence (LGPL) version 3,             #
+#                  copied verbatim in the file "LICENSE"                       #
+################################################################################
+
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+cmake_policy(VERSION 3.14...3.20)
+
+project(FairCMakeModules VERSION 1.0 LANGUAGES NONE)
+
+include(CMakePackageConfigHelpers)
+
+set(PACKAGE_INSTALL_DESTINATION
+  share/cmake/${PROJECT_NAME}-${PROJECT_VERSION}
+)
+
+add_subdirectory(public)
+
+configure_package_config_file(
+  "cmake/${PROJECT_NAME}Config.cmake.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+  INSTALL_DESTINATION "${PACKAGE_INSTALL_DESTINATION}"
+  PATH_VARS CMAKE_INSTALL_PREFIX PACKAGE_INSTALL_DESTINATION)
+write_basic_package_version_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+  VERSION "${PROJECT_VERSION}"
+  COMPATIBILITY SameMajorVersion
```

```suggestion
  COMPATIBILITY AnyNewerVersion
```

I used `SameMajorVersion` before and it was stupid for some reason I forgot, but it is not what one expects from it.

---

done.

Also I don't understand the full problem.

---

I guess it is just the natural meaning of this option, so if you put a version in `find_package`, it will no longer function as a lower version boundary, but this `find_package` statement will only match packages with the same major version. Meaning, if you wanted to support multiple major versions, you have to write multiple `find_package` statements for the same package.